src/main/java/codeu/model/store/basic/MessageStore.java

```diff
@@ -120,5 +143,16 @@ public void addMessage(Message message) {
   /** Sets the List of Messages stored by this MessageStore. */
   public void setMessages(List<Message> messages) {
     this.messages = messages;
+    for (Message message: messages) {
+      UUID authorId = message.getAuthorId();
+      if (messagesByAuthorId.containsKey(authorId)) {
```

Comment:
You can use Java 8 to simplify this:

`messagesByAuthorId.computeIfAbsent(authorId, k -> new ArrayList<Message>()).add(message);`